server/src/data-source.ts

```diff
@@ -0,0 +1,17 @@
+import "reflect-metadata"
+import { DataSource } from "typeorm"
+import { User } from "./entity/User"
+
+export const AppDataSource = new DataSource({
+    type: "postgres",
+    host: "localhost",
+    port: 5432,
+    username: "postgres",
+    password: "123456",
```

Comment:
can you move this information to an env file and read from there?

In the `constants/environments.ts` file, we import all the env and then export them.
So add this to the `.env.example` file and your local `.env`. Then import it into the `environments.ts` file and export it from there. 

```js
interface Environment {
	pi_api_key: string;
	platform_api_url: string;
	DATABASE_HOST: string;
	DATABASE_PORT: string;
	DATABASE_USERNAME: string;
	DATABASE_PASSWORD: string;
}

const env: Environment = {
	pi_api_key: process.env.PI_API_KEY || '',
	platform_api_url: process.env.PLATFORM_API_URL || '',
	DATABASE_HOST: process.env.DATABASE_HOST || 'localhost',
	DATABASE_PORT: process.env.DATABASE_PORT || '5432',
	DATABASE_USERNAME: process.env.DATABASE_USERNAME || 'postgres',
	DATABASE_PASSWORD: process.env.DATABASE_PASSWORD || '123456',
};
```

So that inside this file, you just import `./constants/environments`

now you will have to replace all of the hardcoded data to:
env.DATABASE_HOST
env.DATABASE_PORT
env.DATABASE_USERNAME
env.DATABASE_PASSWORD

Reply:
Okay I will